components/member_list_channel/member_list_channel.tsx

```diff
@@ -135,21 +139,38 @@ export default class MemberListChannel extends React.PureComponent<Props, State>
         this.props.actions.setModalSearchTerm(term);
     }
 
+    toggleShowOnlyAdminUsers= (shouldShowOnlyAdminUsers: boolean): void => {
+        let filters = {};
+        if (shouldShowOnlyAdminUsers) {
+            filters = {
+                roles: [],
+                team_roles: [],
+                channel_roles: [Constants.PERMISSIONS_CHANNEL_ADMIN],
+            };
```

Comment:
Realized just now, this is messing up a few scenarios. Can I know which roles we want to show/filter?

Reply:
It sounds like we want all users that are either a system admin, team admin, or channel admin to show with this filter. I think system admin would go under roles and team admin would go under team roles if I remember how that system works correctly.

Reply:
Unfortunately, using the below is returning mixed results:

```js
filters = {
    roles: [Constants.PERMISSIONS_SYSTEM_ADMIN],
    team_roles: [Constants.PERMISSIONS_TEAM_ADMIN],
    channel_roles: [Constants.PERMISSIONS_CHANNEL_ADMIN],
};
```

![image](https://user-images.githubusercontent.com/17671710/129772805-42cc6fa4-9437-42f0-8c88-f6df1fae9dc9.png)

The `searchProfilesInCurrentTeam` function is returning `UserProfile[]` which contains if the user is a `system_admin` while the `team_admin` role is missing. can we leverage any other selector which returns all the user roles?

Reply:
The team_admin role would likely be on the TeamMembership object for a given user/team, and I think the same thing applies to channel_admin and the ChannelMembership objects. I think we should already have both of those loaded by the time the modal is loaded since they're shown on the dropdown at the right.

It looks like you might already have access to those objects since they come from `getMembersInCurrentTeam` and `getMembersInCurrentChannel`, so maybe take a look at filtering those out there? The only thing that might cause problems with that is if `searchProfilesInCurrentTeam` doesn't return enough users in a page to include the admins, but hopefully that's not the case since it'll require changes to the API and to some server code.